migration/idempotent/001-base.sql

```diff
@@ -2775,11 +2775,65 @@ BEGIN
         RETURN;
     END IF;
 
+    IF NOT _prom_catalog.is_multinode() THEN
+        RETURN QUERY
+        SELECT
+            m.id,
+            m.metric_name,
+            m.table_name::text as table_name,
+            ARRAY
+            (
+                SELECT lkp.key
+                FROM _prom_catalog.label_key_position lkp
+                WHERE lkp.metric_name = m.metric_name
+                ORDER BY lkp.key
+            ) label_keys,
+            coalesce(m.retention_period, _prom_catalog.get_default_retention_period()) as retention_period,
+            _timescaledb_internal.to_interval(d.interval_length) as chunk_interval,
+            x.compressed_interval,
+            x.total_interval,
+            x.before_compression_bytes,
+            x.after_compression_bytes,
+            (x.total_chunk_bytes + pg_total_relation_size(format('%I.%I'::text, h.schema_name, h.table_name)::regclass))::bigint as total_size_bytes,
+            pg_size_pretty(x.total_chunk_bytes + pg_total_relation_size(format('%I.%I'::text, h.schema_name, h.table_name)::regclass)) as total_size,
+            (1.0 - (x.after_compression_bytes::NUMERIC / x.before_compression_bytes::NUMERIC)) * 100 as compression_ratio,
+            x.total_chunks,
+            x.compressed_chunks
+        FROM _prom_catalog.metric m
+        INNER JOIN _timescaledb_catalog.hypertable h ON (m.table_name = h.table_name)
+        INNER JOIN _timescaledb_catalog.dimension d ON (h.id = d.hypertable_id)
+        LEFT OUTER JOIN LATERAL
```

Comment:
(Note: I am requesting changes, but not because of this, this is just discussion)

`LATERAL`  always raises alarm bells for me. In effect the lateral query will be executed once for every row on the left-hand side. I guess in this case it is bounded to #metrics, so it might not be too bad, but if we're going to be returning the results for all metrics, then it could be more efficient to do this differently?

Did you explore using a CTE on the thing in the lateral and grouping by dimension id, and then joining that directly? Something like:

```
WITH dimension_sizes AS (
                SELECT
                d.dimension_id
                count(*) as total_chunks,
                ...
                GROUP BY ds.dimension_id
),
SELECT ...
        FROM _prom_catalog.metric m
        INNER JOIN _timescaledb_catalog.hypertable h ON (m.table_name = h.table_name)
        INNER JOIN _timescaledb_catalog.dimension d ON (h.id = d.hypertable_id)
        LEFT JOIN dimension_sizes ON dimension_sizes.dimension_id = d.id;
```

Reply:
I looked into this a little further. On my test system with 612 metrics it looks as though my suggested query is a little faster (maybe 10%), and uses less memory.

Reply:
how many chunks do you have?